### PR TITLE
Move runtime metadata location

### PR DESCRIPTION
**Pull request by cournape**
This PR makes the hopefully last changes to the runtime format before making it "official":

* `RuntimeVersion` is now based on pep440 instead of semver
* Runtime metadata are now put in `enthought/runtime.json`, not `runtime/runtime.json`
* for python runtimes, the `python_tag` attribute has been added
* in addition to `version` attribute, runtime metadata now have a `language_version` attribute which specifies the language version supported. E.g. for pypy `2.6.1`, the first build of the runtime will have the version `2.6.1+1`, and the python version will be `2.7.10`.
* the dummy runtimes now contain some dummy interpreter so they can be used for jaguar testing

@jvkersch

### CODE REVIEW

File: doc/source/contents.rst.inc [hunk 1]
```diff
@@ -19,3 +19,14 @@ method, this part of the documentation is for you.
    :maxdepth: 2
 
    api
+
+File formats
+------------
+
+If you are looking for information about a specif file format, like our egg or
```
Comment (jvkersch) [inline doc/source/contents.rst.inc]: ... specific ...

File: doc/source/file_formats.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
+.. _file_formats:
+
+Runtimes
+========
+
+Conceptually, runtimes are packages that contain a specific language
+interpreter/compiler. For example, Enthought may provide a python 2.7.10
+runtime, or a pypy-based runtime.
+
+Concretelly, runtimes are simple zipfiles containing the files required for the
```
Comment (jvkersch) [inline doc/source/file_formats.rst]: typo: Concretely

File: doc/source/file_formats.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+to the upstream distribution of the interpreter, e.g. a python runtime on
+windows will be fairly close to a python installed from the python.org msi
+package.
+
+The format is as follows::
+
+    <runtime_file>/enthought/runtime.json
+    <runtime_file>/...
+
+where ``enthought/runtime.json`` directory contains some metadata for Enthought
```
Comment (jvkersch) [inline doc/source/file_formats.rst]: The folder is probably just ``enthought``, right?

**Comment by jvkersch** [comment]
A few questions, but LGTM otherwise.

**Comment by cournape** [comment]
Thanks for the review Joris !

**Pull request merged**